crates/astria-core/src/protocol/transaction/v1alpha1/action_groups.rs

```diff
@@ -0,0 +1,228 @@
+}
+
+impl FeeAssetChangeAction {
+    const ACTION_GROUP: BundlableSudo = BundlableSudo;
+}
+
+impl InitBridgeAccountAction {
+    const ACTION_GROUP: General = General;
+}
+
+impl BridgeLockAction {
+    const ACTION_GROUP: BundlableGeneral = BundlableGeneral;
+}
+
+impl BridgeUnlockAction {
+    const ACTION_GROUP: BundlableGeneral = BundlableGeneral;
+}
+
+impl BridgeSudoChangeAction {
+    const ACTION_GROUP: General = General;
+}
+
+impl FeeChangeAction {
+    const ACTION_GROUP: BundlableSudo = BundlableSudo;
+}
+
+impl_belong_to_group!(
+    SequenceAction,
+    TransferAction,
+    ValidatorUpdate,
+    SudoAddressChangeAction,
+    IbcRelayerChangeAction,
+    Ics20Withdrawal,
+    InitBridgeAccountAction,
+    BridgeLockAction,
+    BridgeUnlockAction,
+    BridgeSudoChangeAction,
+    FeeChangeAction,
+    FeeAssetChangeAction
+);
+
+impl Sealed for Action {}
+impl BelongsToGroup for Action {
+    fn belongs_to_group(&self) -> ActionGroup {
+        match self {
+            Action::Sequence(act) => act.belongs_to_group(),
+            Action::Transfer(act) => act.belongs_to_group(),
+            Action::ValidatorUpdate(act) => act.belongs_to_group(),
+            Action::SudoAddressChange(act) => act.belongs_to_group(),
+            Action::IbcRelayerChange(act) => act.belongs_to_group(),
+            Action::Ics20Withdrawal(act) => act.belongs_to_group(),
+            Action::InitBridgeAccount(act) => act.belongs_to_group(),
+            Action::BridgeLock(act) => act.belongs_to_group(),
+            Action::BridgeUnlock(act) => act.belongs_to_group(),
+            Action::BridgeSudoChange(act) => act.belongs_to_group(),
+            Action::FeeChange(act) => act.belongs_to_group(),
+            Action::FeeAssetChange(act) => act.belongs_to_group(),
+            Action::Ibc(_) => BundlableGeneral.into(), /* Can't implement on action directly
+                                                        * since it lives in a external crate */
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct BundlableGeneral;
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct General;
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct BundlableSudo;
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Sudo;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum ActionGroup {
+    BundlableGeneral(BundlableGeneral),
+    General(General),
+    BundlableSudo(BundlableSudo),
+    Sudo(Sudo),
+}
+
+impl From<BundlableGeneral> for ActionGroup {
+    fn from(val: BundlableGeneral) -> ActionGroup {
+        ActionGroup::BundlableGeneral(val)
+    }
+}
+
+impl From<General> for ActionGroup {
+    fn from(val: General) -> ActionGroup {
+        ActionGroup::General(val)
+    }
+}
+
+impl From<BundlableSudo> for ActionGroup {
+    fn from(val: BundlableSudo) -> ActionGroup {
+        ActionGroup::BundlableSudo(val)
+    }
+}
+
+impl From<Sudo> for ActionGroup {
+    fn from(val: Sudo) -> ActionGroup {
+        ActionGroup::Sudo(val)
+    }
+}
```

Comment:
Note that this was my original suggestion in order to document the group membership for each action.

Doing it via the `Action` enum moves this property away from each action.

Reply:
Well, the associated consts are invisible in rustdoc since they're private, but that's easily remedied I guess.  Likewise, the `BelongsToGroup` trait is private, so it's also invisible.

If we make the `BelongsToGroup` trait `pub` but remove the associated consts, the docs are still pretty unhelpful since each action will have the same generated docs, like:
```
impl BelongsToGroup for BridgeLockAction
fn belongs_to_group(&self) -> ActionGroup
```

We could maybe change the trait to:
```rust
pub trait BelongsToGroup: Sealed {
    const GROUP: ActionGroup;
}
```
so that generated docs look like:
```
impl BelongsToGroup for BridgeLockAction
const GROUP: ActionGroup = ActionGroup::BundlableGeneral
```
It would mean the trait can't be implemented on the `Action` enum, but I think that's a good thing if the whole reason for the trait is to support rustdoc documentation showing which group the given action belongs to, and that is impossible for an enum which can be one of several groups.

Reply:
Yeah, I like that.